cdap-app-fabric/src/main/java/co/cask/cdap/internal/app/runtime/workflow/WorkflowDriver.java

```diff
@@ -178,6 +219,7 @@ private void executeFork(final ApplicationSpecification appSpec, WorkflowForkNod
           public String call() throws Exception {
             Iterator<WorkflowNode> iterator = branch.iterator();
             while (!Thread.currentThread().isInterrupted() && iterator.hasNext()) {
+              checkSuspended();
```

Comment:
should we add this check in the ``executeNode`` method, instead of calling it before every call to ``executeNode()``?